bochs/iodev/usb/usb_uasp.cc

```diff
@@ -0,0 +1,512 @@
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+//  Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public
+//  License along with this library; if not, write to the Free Software
+//  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+/////////////////////////////////////////////////////////////////////////
+
+// Define BX_PLUGGABLE in files that can be compiled into plugins.  For
+// platforms that require a special tag on exported symbols, BX_PLUGGABLE
+// is used to know when we are exporting symbols and when we are importing.
+#define BX_PLUGGABLE
+
+#include "iodev.h"
+
+#if BX_SUPPORT_PCI && BX_SUPPORT_PCIUSB
+#include "usb_common.h"
+#include "hdimage/cdrom.h"
+#include "hdimage/hdimage.h"
+#include "scsi_device.h"
+#include "usb_msd.h"
+
+#define LOG_THIS
+
+// Information Unit Types
+// byte[0] in uasp command and status????
+#define IU_CMD   1
+#define IU_SENSE 3
+#define IU_RESP  4
+#define IU_TMF   5
+#define IU_RRDY  6
+#define IU_WRDY  7
+
+#define IU_CMD_LEN    32
+#define IU_SENSE_LEN  16
+#define IU_RESP_LEN    8
+#define IU_TMF_LEN    16
+#define IU_RRDY_LEN    4
+#define IU_WRDY_LEN    4 
+
+// Task Attribute
+#define UASP_TASK_SIMPLE  0
+#define UASP_TASK_HOQ     1
+#define UASP_TASK_ORDERED 2
+#define UASP_TASK_ACA     4
+
+#define UASP_GET_ACTIVE(m)   (((m) &  1) > 0) // is the request active?
+#define UASP_SET_ACTIVE(m)    ((m) > 0)       // clearing/setting active needs to clear everything else
+#define UASP_GET_STATUS(m)   (((m) &  2) > 0) // have we sent the RRIU or WRIU?
+#define UASP_SET_STATUS(m)    ((m) |  2)      // set that we have sent the RRIU or WRIU
+#define UASP_GET_CMND(m)     (((m) &  4) > 0) // have we received and processed the command yet?
+#define UASP_SET_CMND(m)      ((m) |  4)      // set that we have received and processed the command
+#define UASP_GET_COMPLETE(m) (((m) &  8) > 0) // is the datain/dataout transfer complete?
+#define UASP_SET_COMPLETE(m)  ((m) |  8)      // set that the datain/dataout transfer is complete
+#define UASP_GET_DIR(m)      (((m) &  0x0000FF00) >> 8)
+#define UASP_SET_DIR(m,d)    (((m) & ~0x0000FF00) | ((d) << 8))
+
+struct S_UASP_COMMAND {
+	Bit8u  id;         // Information Unit Type
+	Bit8u  rsvd0;      //
+	Bit16u tag;        // big endian
+	Bit8u  prio_attr;  // Task Attribute
+	Bit8u  rsvd1;      //
+	Bit8u  len;        // len (if more than 16 bytes) (i.e.: if command len is 20 bytes, this field is 4)  (bottom 2 bits are reserved)
+	Bit8u  rsvd2;      //
+	Bit64u lun;        // eight byte lun
+  Bit8u  com_block[16]; // command block
+};
+
+struct S_UASP_STATUS {
+	Bit8u  id;         // usually IU_SENSE ?
+	Bit8u  rsvd0;
+	Bit16u tag;        // big endian
+	Bit16u stat_qual;  // big endian
+	Bit8u  status;
+	Bit8u  resv1[7];
+	Bit16u len;        // big endian
+  Bit8u  sense[18];
```

Comment:
inconsistent indentation, is it intentional ?

Reply:
done. ditto.